internal/work/dispatcher.go

```diff
@@ -144,15 +144,12 @@ func (d *Dispatcher) Connect() error {
 				}
 				event.MessageID = eventMessageID
 
-				switch ipc.WorkerEventName(eventName) {
-				case ipc.WorkerEventNameWorking:
-					eventMessage, ok := s.Body[2].(string)
-					if !ok {
-						log.Errorf("cannot convert %T to string", s.Body[2])
-						continue
-					}
-					event.Message = eventMessage
+				eventMessageData, ok := s.Body[2].(map[string]string)
+				if !ok {
+					log.Errorf("cannot convert %T to map[string]string", s.Body[2])
+					continue
 				}
+				event.Message = eventMessageData["message"]
```

Comment:
Similarly, the `WorkerEvent` struct should change to have a `data map[string]string` field.